integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -351,7 +351,7 @@ object CommonUtil {
     if (tempList.length != originListInfo.size) {
       sys.error("The total number of elements in new list must equal to original list!")
     }
-    if (!originListInfo.sameElements(tempList)) {
+    if (originListInfo.toSet!=tempList.toSet) {
```

Comment:
According to the error message below, it requires `originListInfo` contains `tempList`, so the condition should be like `!(tempList diff originListInfo).isEmpty`